src/js/components/search/filters/programSource/SourceSelectFilter.jsx

```diff
@@ -0,0 +1,166 @@
+/**
+ * SourceSelectFilter.jsx
+ * Created by Lizzie Salita 6/7/19
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { isEqual, differenceWith } from 'lodash';
+
+import Autocomplete from 'components/sharedComponents/autocomplete/Autocomplete';
+
+const propTypes = {
+    label: PropTypes.string,
+    code: PropTypes.string,
+    characterLimit: PropTypes.number,
+    required: PropTypes.bool,
+    options: PropTypes.array,
+    selectedSources: PropTypes.array,
+    updateComponent: PropTypes.func,
+    dirtyFilters: PropTypes.symbol
+};
+
+const defaultProps = {
+    // TODO - Lizzie: remove mock data
+    options: [
+        {
+            code: '020',
+            name: 'Department of the Treasury (TREAS)'
+        },
+        {
+            code: '014',
+            name: 'Department of the Interior (DOI)'
+        },
+        {
+            code: '068',
+            name: 'Environmental Protection Agency (EPA)'
+        },
+        {
+            code: '9876'
+        }
+    ],
+    selectedSources: [],
+    required: false
+};
+
+export class SourceSelectFilter extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            searchString: '',
+            autocompleteOptions: [],
+            noResults: false
+        };
+
+        this.handleTextInput = this.handleTextInput.bind(this);
+        this.clearAutocompleteSuggestions = this.clearAutocompleteSuggestions.bind(this);
+        this.selectSourceComponent = this.selectSourceComponent.bind(this);
+        this.timeout = null;
+    }
+
+    parseAutocompleteOptions(options) {
+        const values = [];
+        if (options && options.length > 0) {
+            options.forEach((item) => {
+                values.push({
+                    title: item.code,
+                    subtitle: item.name || '',
+                    data: item
+                });
+            });
+        }
+
+        this.setState({
+            autocompleteOptions: values
+        });
+    }
+
+    queryAutocomplete(input) {
+        this.setState({
+            noResults: false
+        });
+
+        // Only search if input is 1 or more characters
+        if (input.length >= 1) {
+            this.setState({
+                searchString: input
+            });
+
+            let autocompleteOptions = [];
+
+            const matches = this.props.options.filter((source) => (source.code.includes(this.state.searchString)));
+            // Filter out any selectedSources that may be in the result set
+            if (this.props.selectedSources && this.props.selectedSources.length > 0) {
```

Comment:
I realized we only need the first filter. We don't need to prevent users from picking a value that's already been selected, because the component (AID for example) is part of the larger Federal Account number and could repeat in different filters (e.g. `020-1234`, `020-5678`).